src/plugins/intel_gpu/src/plugin/common_utils.cpp

```diff
@@ -138,6 +138,16 @@ void convert_and_copy(const cldnn::memory::ptr src, ov::ITensor const* dst, cons
     return ::convert_and_copy(src_ptr, src_et, dst_ptr, dst_et, size, src->get_layout());
 }
 
+void convert_and_copy(const cldnn::memory::ptr src, cldnn::memory::ptr dst, cldnn::stream& stream) {
+    const bool blocking = true;
+
+    cldnn::mem_lock<uint8_t, cldnn::mem_lock_type::read> src_lock(src, stream);
+    std::unique_ptr<cldnn::mem_lock<uint8_t>> dst_lock = nullptr;
+
+    const void* src_ptr = src_lock.data();
+    dst->copy_from(stream, src_ptr, blocking);
```

Comment:
This call just copies the memory without data conversion; but if the data types are different, we need to convert it before storing to dst memory